src/Action/InitElasticSearch.php

```diff
@@ -1,27 +1,34 @@
 <?php
-/**  
+/**
  * This program is free software; you can redistribute it and/or
  * modify it under the terms of the GNU General Public License
  * as published by the Free Software Foundation; under version 2
  * of the License (non-upgradable).
- * 
+ *
  * This program is distributed in the hope that it will be useful,
  * but WITHOUT ANY WARRANTY; without even the implied warranty of
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  * GNU General Public License for more details.
- * 
+ *
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
- * 
+ *
  * Copyright (c) 2018 (original work) Open Assessment Technologies SA;
- *               
- * 
  */
+
+declare(strict_types=1);
+
 namespace oat\tao\elasticsearch\Action;
 
+use Elasticsearch\Common\Exceptions\BadRequest400Exception;
 use oat\tao\elasticsearch\ElasticSearch;
 use common_report_Report as Report;
+use oat\tao\elasticsearch\Watcher\IndexDocumentFactory;
+use oat\tao\model\search\index\IndexService;
+use oat\tao\elasticsearch\IndexUpdater;
+use oat\tao\model\search\index\IndexUpdaterInterface;
+use oat\tao\model\search\strategy\GenerisSearch;
```

Comment:
i think we a adding more coupling inside the library by using models from tao-core, 
i was hoping to have here a interface that we use inside tao core not the other way around